### PR TITLE
 Add Rules engine to CLI

### PR DESCRIPTION
**Pull request by shaswa**
- This adds rules engine to CLI
- Can List and sync rules with Insights for a given org
  - Each rule would be in a separate file
  - List outputs both OPA checks and rules
  - Sync only syncs either OPA checks or rules. Default is checks, passing `--rules` flag syncs rules instead
- Can perform dry run on the sync operation

TODO
- Update documentation with examples
- Add tests

### CODE REVIEW

File: pkg/rules/rules.go [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+		if rule.Cluster != "" {
+			value := fmt.Sprintf("Cluster: %s", rule.Cluster)
+			rulesNode.AddNode(value)
+		}
+		if rule.Context != "" {
+			value := fmt.Sprintf("Context: %s", rule.Context)
+			rulesNode.AddNode(value)
+		}
+		if rule.Repository != "" {
+			value := fmt.Sprintf("Cluster: %s", rule.Repository)
```
Comment (baderbuddy) [inline pkg/rules/rules.go:142]: Copypasta, I'm guessing tthis is supposed to be Repostiory:

File: pkg/rules/rules.go [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+			rulesNode.AddNode(value)
+		}
+		if rule.ReportType != "" {
+			value := fmt.Sprintf("Cluster: %s", rule.ReportType)
```
Comment (baderbuddy) [inline pkg/rules/rules.go:146]: Another copy error for Report Type

**Review by baderbuddy — COMMENTED** [review]
Found a couple minor issues, I'll look more at this later

**Review by baderbuddy — APPROVED** [review]
Overall I like the structure. I had originally created that Insights package to encapsulate all of the HTTP calls to Insights, but that really doesn't seem very useful anymore.

The one thing I'm still going back and forth on and I don't know if this is a good idea. If we a force a certain reserved name (like rules), for a folder, then we could sync both OPA checks and Rules at the same time. Maybe that's not actually valuable though.

**Comment by shaswa** [comment]
> Overall I like the structure. I had originally created that Insights package to encapsulate all of the HTTP calls to Insights, but that really doesn't seem very useful anymore.
> 
> The one thing I'm still going back and forth on and I don't know if this is a good idea. If we a force a certain reserved name (like rules), for a folder, then we could sync both OPA checks and Rules at the same time. Maybe that's not actually valuable though.

I wanted to enforce a name like that too for the folders but I'm not sure either.

**Comment by shaswa** [comment]
Updated to sync both checks and rules at the same time by enforcing a naming convention

All checks should be in folder `checks` and all rules in a folder `rules`.  The directory passed into the command is the parent directory of these two folders

**Directory structure**

Parent directory
> checks
> > check1folder
> >  check2folder

> rules
> > rule1file
> >  rule2file
> >  rule3file

**Review by baderbuddy — APPROVED** [review]
Ohh, we should definitely make this a major release when we cut it. But that is a really nice way of handling this.

**Comment by shaswa** [comment]
Thank you, Bader!

**Pull request merged**